src/libpgagroal/configuration.c

```diff
@@ -178,6 +178,46 @@ pgagroal_init_configuration(void* shm)
    return 0;
 }
 
+/**
+ *
+ */
+void
+pgagroal_check_directory_path(char** path, const char* directory_path, const char* default_path, const char* fallback_path)
+{
+   char temp_path[MAX_PATH];
+
+   snprintf(temp_path, MAX_PATH, "%s/%s", directory_path, default_path + strlen(PGAGROAL_DEFAULT_CONFIGURATION_PATH));
+   if (*path == NULL && access(temp_path, F_OK) == 0)
```

Comment:
I would first check if `path` is 'NULL` or not, then in case it is, do the `snprintf` and other checks.